client.go

```diff
@@ -183,7 +185,20 @@ func NewClient(config Config) (Marathon, error) {
 	client.cluster = cluster
 	client.httpClient = &http.Client{
 		Timeout: (time.Duration(config.RequestTimeout) * time.Second),
+		// a copy of http.DefaultTransport
+		Transport: &http.Transport{
+			Proxy: http.ProxyFromEnvironment,
+			Dial: (&net.Dialer{
+				Timeout:   30 * time.Second,
+				KeepAlive: 30 * time.Second,
+			}).Dial,
+			TLSClientConfig: &tls.Config{
+				InsecureSkipVerify: config.SkipTLSVerify,
+			},
+			TLSHandshakeTimeout: 10 * time.Second,
+		},
```

Comment:
Could it be easier just to have an option to specify your own `httpClient` in config and fall-back to default. In this case user can provide all necessary options (like self-signed certificates) and we just need to provide an example how to do it.